### PR TITLE
Scale line width3305

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
Addresses suggestions in #3305 

#### What does this implement/fix? Explain your changes.
This no longer scales bond widths by default.  There's a new option to turn line width scaling on if required.
getDrawLineWidth() now returns a double.

#### Any other comments?
As a side order, the size of the ellipse for atom highlights was altered to hopefully a more sensible size.
Tested on Mac only.

### CODE REVIEW

File: Code/GraphMol/MolDraw2D/test1.cpp [hunk 1]
```diff
@@ -2218,6 +2216,9 @@ void testGithub1322() {
 }
 
 void testGithub565() {
+  // this is expected not to give a good looking drawing.  There is now
+  // There is now a drawOptions().fixedBondLength achieve this.  I think
+  // this test should be withdrawn.
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/test1.cpp:2221]: agreed

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 2]
```diff
@@ -1727,6 +1730,8 @@ void MolDraw2D::drawHighlightedBonds(
     const map<int, int> &highlight_linewidth_multipliers,
     const map<int, double> *highlight_radii) {
   int orig_lw = lineWidth();
+  bool orig_lws = drawOptions().scaleBondWidth;
+  drawOptions().scaleBondWidth = true;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:1734]: I'm curious why you're doing this. If we're not scaling bond widths it doesn't seem to make sense to scale the highlights either.

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 3]
```diff
@@ -113,7 +112,13 @@ void MolDraw2D::doContinuousHighlighting(
   PRECONDITION(activeMolIdx_ >= 0, "bad active mol");
 
   int orig_lw = lineWidth();
+  bool orig_lws = drawOptions().scaleBondWidth;
+  drawOptions().scaleBondWidth = true;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:116]: same comment here that I made below: it seems to me that it's confusing/counter-intuitive to scale the highlighting if we aren't scaling the bond widths. Is there a particular example that led to this?

File: Code/GraphMol/MolDraw2D/Wrap/rdMolDraw2D.cpp [hunk 4]
```diff
@@ -619,6 +619,8 @@ BOOST_PYTHON_MODULE(rdMolDraw2D) {
       .def_readwrite(
           "bondLineWidth", &RDKit::MolDrawOptions::bondLineWidth,
           "if positive, this overrides the default line width for bonds")
+      .def_readwrite("scaleBondWidth", &RDKit::MolDrawOptions::scaleBondWidth,
+                     "Scales the width of drawn bonds.")
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/Wrap/rdMolDraw2D.cpp:623]: perhaps: "scales the width of drawn bonds using the image scaling"?

**Review by greglandrum — CHANGES_REQUESTED** [review]
Upon first read-through I don't agree with the decision here to always scale the highlight bond widths. Before merging I'd like to either change this or to understand why you chose that behavior.

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp (hunk 2 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/MolDraw2D.cpp:1734]: test3_5.svg and test15_1.svg look horrible without.  But you're right, it is inconsistent so I will look at other ways of mending them.

File: Code/GraphMol/MolDraw2D/test1.cpp [hunk 5]
```diff
@@ -371,6 +371,7 @@ void test3() {
     }
 #endif
     {
+      std::cout << "test3_5" << std::endl;
```
[the remark this answers is not in the document]
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/test1.cpp:374]: Sorry.

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp (hunk 3 above)
Comment (DavidACosgrove) [reply Code/GraphMol/MolDraw2D/MolDraw2D.cpp]: After off-line discussion, drawOptions().scaleHighlightBondWidth has been added.

**Comment by greglandrum** [comment]
@DavidACosgrove : the CI failures here are all real and related to the changes in the PR. Can you please take a look?

File: Code/GraphMol/MolDraw2D/DrawTextFT.cpp [hunk 6]
```diff
@@ -112,12 +112,14 @@ void DrawTextFT::setFontFile(const std::string &font_file) {
   face_ = nullptr;
   // take the first face
   if (!font_file_.empty()) {
+    std::cerr << "loading font from " << font_file << std::endl;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawTextFT.cpp:115]: please remove debugging output

File: Code/GraphMol/MolDraw2D/DrawTextFT.cpp [hunk 7]
```diff
@@ -112,12 +112,14 @@ void DrawTextFT::setFontFile(const std::string &font_file) {
...
     int err_code = FT_New_Face(library_, font_file_.c_str(), 0, &face_);
     if (err_code != FT_Err_Ok) {
       throw std::runtime_error(std::string("Font file ") + font_file_ +
                                std::string(" not found."));
     }
   } else {
+    std::cerr << "loading built in font" << std::endl;
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/DrawTextFT.cpp:122]: please remove debugging output

**Review by greglandrum — APPROVED** [review]

**Pull request merged**